src/runtime/network_monitor.cpp

```diff
@@ -0,0 +1,149 @@
+#include "network_monitor.hpp"
+
+#if UNITY_ANDROID
+#include <android/log.h>
+#endif
+
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <unistd.h>
+#include <thread>
+#include <chrono>
+
+namespace runtime
+{
+  std::shared_ptr<NetworkMonitor> NetworkMonitor::create()
+  {
+#if UNITY_ANDROID
+    return std::make_shared<AndroidNetworkMonitor>();
+#elif UNITY_OSX
+    return std::make_shared<MacOSNetworkMonitor>();
+#else
+    // For other platforms, use the socket-based implementation as default
+    return std::make_shared<SocketBasedNetworkMonitor>();
+#endif
+  }
+
+  // SocketBasedNetworkMonitor implementation
```

Comment:
Use a name to represent that works for all platforms, and can be safely downgrade.